modules/base/rendering/renderableprism.cpp

```diff
@@ -0,0 +1,354 @@
+        std::optional<glm::vec3> color [[codegen::color()]];
+
+        // [[codegen::verbatim(LengthInfo.description)]]
+        std::optional<float> length;
+    };
+#include "renderableprism_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderablePrism::Documentation() {
+    documentation::Documentation doc = codegen::doc<Parameters>("base_renderable_prism");
+    return doc;
+}
+
+RenderablePrism::RenderablePrism(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _nShapeSegments(SegmentsInfo, 6, 3, 32)
+    , _nLines(LinesInfo, 6, 0, 32)
+    , _radius(RadiusInfo, 10.f, 0.f, 3.0e12f)
+    , _lineWidth(LineWidthInfo, 1.f, 1.f, 20.f)
+    , _lineColor(LineColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _length(LengthInfo, 20.f, 1.f, 3.0e12f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _radius.setViewOption(properties::Property::ViewOptions::Logarithmic);
+    _lineColor.setViewOption(properties::Property::ViewOptions::Color);
+    _length.setViewOption(properties::Property::ViewOptions::Logarithmic);
+
+    _nShapeSegments.onChange([&]() { _prismIsDirty = true; });
+    _nLines.onChange([&]() { _prismIsDirty = true; });
+    _radius.onChange([&]() { _prismIsDirty = true; });
+    _length.onChange([&]() { _prismIsDirty = true; });
+
+    _nShapeSegments = p.segments;
+    _nLines = p.lines.value_or(_nShapeSegments);
+    _radius = p.radius.value_or(_radius);
+    _lineWidth = p.lineWidth.value_or(_lineWidth);
+    _lineColor = p.color.value_or(_lineColor);
+    _length = p.length.value_or(_length);
+
+    addProperty(_nShapeSegments);
+    addProperty(_nLines);
+    addProperty(_radius);
+    addProperty(_lineWidth);
+    addProperty(_lineColor);
+    addProperty(_length);
+    addProperty(_opacity);
+}
+
+bool RenderablePrism::isReady() const {
+    return _shader != nullptr;
+}
+
+void RenderablePrism::initialize() {
+    updateVertexData();
+}
+
+void RenderablePrism::initializeGL() {
+    initializeShader();
+    ghoul::opengl::updateUniformLocations(*_shader, _uniformCache, UniformNames);
+
+    glGenVertexArrays(1, &_vaoId);
+    glGenBuffers(1, &_vboId);
+    glGenBuffers(1, &_iboId);
+
+    glBindVertexArray(_vaoId);
+
+    updateBufferData();
+
+    glEnableVertexAttribArray(_locVertex);
+    glVertexAttribPointer(_locVertex, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), nullptr);
+    glBindVertexArray(0);
+}
+
+void RenderablePrism::deinitializeGL() {
+    global::renderEngine->removeRenderProgram(_shader.get());
+    _shader = nullptr;
+
+    glDeleteVertexArrays(1, &_vaoId);
+    _vaoId = 0;
+
+    glDeleteBuffers(1, &_vboId);
+    _vboId = 0;
+
+    glDeleteBuffers(1, &_iboId);
+    _iboId = 0;
+}
+
+void RenderablePrism::updateVertexData() {
+    _vertexArray.clear();
+    _indexArray.clear();
+
+    // Get unit circle vertices on the XY-plane
+    std::vector<float> unitVertices;
```

Comment:
With the change above this would become:
```cpp
std::vector<float> unitVertices = unitCircleVertices(_nShapeSegments);
std::vector<float> unitVerticesLines = unitCircleVertices(_nLines);
```

And the `reserve` functions can move into the `unitCircleVertices` function itself